lib/src/features/home/presentation/pages/filter_page/widgets/colors_section.dart

```diff
@@ -0,0 +1,106 @@
+import 'package:ecommerce_app/core/l10n/l10n.dart';
+import 'package:ecommerce_app/core/theme/colors.dart';
+import 'package:ecommerce_app/core/theme/text_styles.dart';
+import 'package:ecommerce_app/src/features/home/presentation/pages/filter_page/bloc/filter_bloc.dart';
+import 'package:ecommerce_app/src/features/home/presentation/pages/filter_page/bloc/filter_event.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+class ColorsSection extends StatefulWidget {
+  const ColorsSection({super.key});
+
+  @override
+  State<ColorsSection> createState() => _ColorsSectionState();
+}
+
+class _ColorsSectionState extends State<ColorsSection> {
+  @override
+  void initState() {
+    super.initState();
+    context.read<FilterBloc>().add(LoadColorsEvent());
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final colors = context.select<FilterBloc, List<Color>>(
+      (bloc) => bloc.state.colors,
+    );
+    final selectedColors = context.select<FilterBloc, List<Color>>(
+      (bloc) => bloc.state.selectedColors,
+    );
+    final colorsErrorMessage = context.select<FilterBloc, String>(
+      (bloc) => bloc.state.colorsErrorMessage,
+    );
+
+    if (colorsErrorMessage.isNotEmpty) {
+      return Center(child: Text(colorsErrorMessage));
+    }
+
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        Text(
+          context.localization.filterPageColorsText,
+          style: FilterPageTextStyles.sectionNameTextStyle,
+        ),
+        const SizedBox(height: 8),
+        Row(
+          mainAxisAlignment: MainAxisAlignment.spaceBetween,
+          children: colors.map((color) {
+            if (color == AppColors.blueColor) {
```

Comment:
Looks like this padding is not related to the color, it's related to the index. Maybe change it with `if (i == 0)` ?